aspnetcore/security/enforcing-ssl.md

```diff
@@ -26,44 +28,58 @@ This document shows how to:
 
-The following code calls [AddHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpsredirectionservicesextensions.addhttpsredirection) to configure middleware options:
+> [!WARNING]
+>A port must be available for the middleware to redirect to HTTPS. If no port is available, redirection to HTTPS does not occur. The HTTPS port can be specified by setting:
+>
+>* `HttpsRedirectionOptions.HttpsPort` or
+>* The `ASPNETCORE_HTTPS_PORT` environment variable, or 
+>* An HTTPS url in *launchsettings.json*
+
+The following code highlighted code calls [AddHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpsredirectionservicesextensions.addhttpsredirection) to configure middleware options:
 
 [!code-csharp[](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=14-99)]
 
+It's not necessary to call `AddHttpsRedirection`. Call `AddHttpsRedirection` to change the ` HttpsPort` or ` RedirectStatusCode`;
+
 The preceding highlighted code:
 
-* Sets [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode) to `Status307TemporaryRedirect`, which is the default value. Production apps should call [UseHsts](#hsts).
+* Sets [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode) to `Status307TemporaryRedirect`, which is the default value.
 * Sets the HTTPS port to 5001. The default value is 443.
 
 The following mechanisms set the port automatically:
 
 * The middleware can discover the ports via [IServerAddressesFeature](/dotnet/api/microsoft.aspnetcore.hosting.server.features.iserveraddressesfeature) when the following conditions apply:
-  - Kestrel or HTTP.sys is used directly with HTTPS endpoints (also applies to running the app with Visual Studio Code's debugger).
-  - Only **one HTTPS port** is used by the app.
+  * Kestrel or HTTP.sys is used directly with HTTPS endpoints (also applies to running the app with Visual Studio Code's debugger).
+  * Only **one HTTPS port** is used by the app.
 * Visual Studio is used:
-  - IIS Express has HTTPS enabled.
-  - *launchSettings.json* sets the `sslPort` for IIS Express.
+  * IIS Express has HTTPS enabled.
+  * *launchSettings.json* sets the `sslPort` for IIS Express.
 
 > [!NOTE]
 > When an app is run behind a reverse proxy (for example, IIS, IIS Express), `IServerAddressesFeature` isn't available. The port must be manually configured. When the port isn't set, requests aren't redirected.
 
 The port can be configured by setting the [https_port Web Host configuration setting](xref:fundamentals/host/web-host#https-port):
 
-**Key**: https_port
-**Type**: *string*
-**Default**: A default value isn't set.
-**Set using**: `UseSetting`
+**Key**: https_port 
```

Comment:
Each of these (Lines 79-82) require *two spaces* at the end to render on separate lines (two spaces results in a `<br>` for each line).